lib/web/purchases_flutter_web.dart

```diff
@@ -0,0 +1,428 @@
+      throw PlatformException(
+        code: _configurationErrorCode,
+        message: 'Purchases SDK not configured. Call configure first.',
+      );
+    }
+  }
+
+  Future<void> _setLogLevel(dynamic arguments) async {
+    final logLevel = arguments['level'] as String;
+    final jsLogLevel = _convertLogLevel(logLevel);
+    _callStaticMethod('setLogLevel', [jsLogLevel]);
+  }
+
+  String _convertLogLevel(String logLevel) => switch (logLevel.toLowerCase()) {
+    'verbose' => 'VERBOSE',
+    'debug' => 'DEBUG',
+    'info' => 'INFO',
+    'warn' => 'WARN',
+    'error' => 'ERROR',
+    _ => 'SILENT',
+  };
+
+  bool _isConfigured() => _callStaticMethod('isConfigured', []) as bool;
+
+  Future<Map<String, dynamic>> _getCustomerInfo() async =>
+      await _getMapFromInstanceMethod('getCustomerInfo', []);
+
+  Future<Map<String, dynamic>> _getOfferings() async =>
+      await _getMapFromInstanceMethod('getOfferings', []);
+
+  Future<Map<String, dynamic>?> _getCurrentOfferingForPlacement(
+    dynamic arguments,
+  ) async {
+    final placementIdentifier = arguments['placementIdentifier'] as String;
+    return await _getNullableMapFromInstanceMethod(
+        'getCurrentOfferingForPlacement',
+        [placementIdentifier],
+    );
+  }
+
+  Future<String> _getAppUserID() async {
+    final purchases = _getInstance();
+    return purchases.callMethod('getAppUserId'.toJS).dartify().toString();
+  }
+
+  Future<Map<String, dynamic>> _logIn(dynamic arguments) async {
+    final appUserId = arguments['appUserID'] as String;
+    final result = await _getMapFromInstanceMethod('logIn', [appUserId]);
+    return result;
+  }
+
+  Future<Map<String, dynamic>> _logOut() async =>
+      await _getMapFromInstanceMethod('logOut', []);
+
+  Future<Map<String, dynamic>> _purchasePackage(dynamic arguments) async {
+    final packageIdentifier = arguments['packageIdentifier'] as String;
+    final options = {
+      'packageIdentifier': packageIdentifier,
+      'presentedOfferingContext': arguments['presentedOfferingContext'],
+    };
+    return await _getMapFromInstanceMethod('purchasePackage', [options]);
+  }
+
+  Future<Map<String, dynamic>> _restorePurchases() async =>
+    // Web SDK doesn't support restore purchases, but returns current customer info
+    // to match behavior with other platforms
+    await _getCustomerInfo();
+
+  Future<void> _close() async {
+    _getInstance().callMethod('close'.toJS);
+  }
+
+  Future<bool> _isAnonymous() async {
+    final purchases = _getInstance();
+    return purchases.callMethod('isAnonymous'.toJS).dartify() as bool;
+  }
+
+  Future<bool> _isSandbox() async {
+    final purchases = _getInstance();
+    return purchases.callMethod('isSandbox'.toJS).dartify() as bool;
+  }
+
+  Future<Map<String, dynamic>> _checkTrialOrIntroductoryPriceEligibility(
+    dynamic arguments,
+  ) async {
+    // Web SDK doesn't support trial/intro eligibility checking
+    // Return unknown status for all products
```

Comment:
I wonder, would it be preferable to fail the function in this case? I don't have a clear opinion but I *think* it makes more sense

Reply:
Yeah we did it this way because it's how it behaves in Android so I think it's better to be consistent there, but I could totally see your point as well. Lmk if you feel strongly about it though!